runner/utils/driver/istio/get.go

```diff
@@ -269,12 +269,60 @@ func (i *Istio) GetServiceRoutes(ctx context.Context, projectID string) (map[str
 
 	for _, service := range services.Items {
 		serviceID := service.Labels["app"]
-		routes := make(model.Routes, len(service.Spec.Http)+len(service.Spec.Tcp))
+		routes := make(model.Routes, 0)
 
-		for i, route := range service.Spec.Http {
+		for _, route := range service.Spec.Http {
 
 			// Generate the targets
 			targets := make([]model.RouteTarget, len(route.Route))
+			matchers := make([]*model.Matcher, len(route.Match))
+			for i, match := range route.Match {
+				tempMatcher := new(model.Matcher)
+
+				if match.Uri != nil && match.Uri.GetMatchType() != nil {
+					tempMatcher.URL = new(model.HTTPMatcher)
+					tempMatcher.URL.IgnoreCase = match.IgnoreUriCase
+					if exact := match.Uri.GetExact(); exact != "" {
+						tempMatcher.URL.Type = model.RouteHTTPMatchTypeExact
+						tempMatcher.URL.Value = exact
+					}
+					if prefix := match.Uri.GetPrefix(); prefix != "" {
+						tempMatcher.URL.Type = model.RouteHTTPMatchTypePrefix
+						tempMatcher.URL.Value = prefix
+					}
+					if regex := match.Uri.GetRegex(); regex != "" {
+						tempMatcher.URL.Type = model.RouteHTTPMatchTypeRegex
+						tempMatcher.URL.Value = regex
+					}
+				}
+
+				tempMatcher.Headers = make([]*model.HTTPMatcher, 0)
+				for headerKey, headerValue := range match.Headers {
+					tempHeader := new(model.HTTPMatcher)
+					tempHeader.Key = headerKey
+
+					if exact := headerValue.GetExact(); exact != "" {
+						tempHeader.Type = model.RouteHTTPMatchTypeExact
+						tempHeader.Value = exact
+					}
+					if prefix := headerValue.GetPrefix(); prefix != "" {
+						tempHeader.Type = model.RouteHTTPMatchTypePrefix
+						tempHeader.Value = prefix
+					}
+					if regex := headerValue.GetRegex(); regex != "" {
+						tempHeader.Type = model.RouteHTTPMatchTypeRegex
+						tempHeader.Value = regex
+					}
+
+					if tempHeader.Value == "" || tempHeader.Type == "" {
+						tempHeader.Type = model.RouteHTTPMatchTypeCheckPresence
+					}
```

Comment:
I think we should do this first because headerValue itself can be nil

Reply:
A header value cannot be nil,
If you try to assign nil to header value, ISTIO throws an error